File: cpp/estrutura_sequencial/medidas.cpp
/* Problema "medidas"
 Fazer um programa para ler três medidas A, B e C. Em seguida, calcular e mostrar (imprimir os dados
com quatro casas decimais):
 a) a área do quadrado que tem lado A
 b) a área do triângulo retângulo que base A e altura B
 c) a área do trapézio que tem bases A e B, e altura C
*/

#include <iostream>
#include <iomanip>

using namespace std;

int main()
{
  double a, b, c, square_area, triangle_area, trapeze_area;

  cout << "Digite a medida A: ";
  cin >> a;

  cout << "Digite a medida B: ";
  cin >> b;

  cout << "Digite a medida C: ";
  cin >> c;

  square_area = a * a;
  triangle_area = a * b / 2.0;
  trapeze_area = ((a + b) * c) / 2.0;

  cout << fixed << setprecision(4);
  cout << "ÁREA DO QUADRADO = " << square_area << endl;
  cout << "ÁREA DO TRIÂNGULO = " << triangle_area << endl;
  cout << "ÁREA DO TRAPÉZIO = " << trapeze_area << endl;

  return 0;
}

File: csharp/Source/Estruturas/Sequencial/Medidas.cs
namespace Source.Estruturas.Sequencial;

/**
 * Problema "medidas"
 * Fazer um programa para ler três medidas A, B e C.
 * Em seguida, calcular e mostrar (imprimir os dados
 * com quatro casas decimais):
 * a) a área do quadrado que tem lado A
 * b) a área do triângulo retângulo que base A e altura B
 * c) a área do trapézio que tem bases A e B, e altura C
 */

internal static class Medidas
{
    public static void Execute()
    {

        Console.Write("Digite a medida A: ");
        double a = double.Parse(Console.ReadLine()!);

        Console.Write("Digite a medida B: ");
        double b = double.Parse(Console.ReadLine()!);

        Console.Write("Digite a medida C: ");
        double c = double.Parse(Console.ReadLine()!);

        double squareArea = a * a;
        double triangleArea = a * b / 2.0;
        double trapezeArea = ((a + b) * c) / 2.0;

        Console.WriteLine("ÁREA DO QUADRADO = {0:F4}", squareArea);
        Console.WriteLine("ÁREA DO TRIÂNGULO = {0:F4}", triangleArea);
        Console.WriteLine("ÁREA DO TRAPÉZIO = {0:F4}", trapezeArea);
    }
}